connectivity/mbedtls/tests/TESTS/mbedtls/selftest/main.cpp

```diff
@@ -65,6 +66,10 @@ MBEDTLS_SELF_TEST_TEST_CASE(mbedtls_sha512_self_test)
 MBEDTLS_SELF_TEST_TEST_CASE(mbedtls_entropy_self_test)
 #endif
 
+#if defined(MBEDTLS_TIMING_C)
```

Comment:
We need a test config json to enable TIMING_C and TIMING_ALT and SELF_TEST. This is adding dead code otherwise. We also need to update CI to use the test config json for `mbedtls/selftest`. We can also take the opportunity to disable SELF_TEST by default.

Reply:
I've added a json file for self tests and disabled MBEDTLS_SELF_TEST by default.